### PR TITLE
Add new method ApplyTag

### PR DESCRIPTION
**Pull request by haoxins**
## What type of PR is this?
- [ ] bug
- [x] feature
- [ ] enhancement

#### Description:

```
ApplyTag applies the given tag to the graph.

1. If the tag does not exist, it will be created.
2. If the tag exists, it will be checked if the fields are the same.
2.1 If not, the new fields will be added.
2.2 If the field type is different, it will return an error.
2.3 If a field exists in the graph but not in the given tag,
it will be removed.

Notice:
We won't change the field type because it has
unexpected behavior for the data.
```

### CODE REVIEW

**Comment by wey-gu** [comment]
This sugar on Client Side is so very inspiring.

We should consider do similar things on all Clients.

**Comment by Nicole00** [comment]
This is a very convenient interface that can provide users with an encapsulated and convenient interface. Thanks for your idea. 
I'm thinking whether it's appropriate to put such syntactic sugar inside the client? Are their application scenarios general enough? After all, the client serves as an SDK rather than an ORM framework. Maybe we can add these sugar in another go file to differentiate from session or session_pool. What's your idea? @haoxin @wey-gu

**Comment by haoxins** [comment]
> Maybe we can add these sugar in another go file to differentiate from session or session_pool.

The idea of `applying xxx` comes from the cloud-native community (or K8s), I think it is the basic requirement of schema management.

The benefit is that we don't need to pass `session_pool` if we add this into the `session_pool` struct.

But I agree that it does not belong to `session_pool/session`.
Do you have an idea for the name of this?

**Comment by haoxins** [comment]
`SchemaManager`?

**Comment by Nicole00** [comment]
> `SchemaManager`?

LGTM. cc @wey-gu

**Comment by wey-gu** [comment]
Agree with @Nicole00 that it's strange to be placed under the session_pool tent, `SchemaManager` sounds great to me, too :D

**Comment by haoxins** [comment]
done cc @Nicole00 @wey-gu

**Review by wey-gu — APPROVED** [review]

**Comment by wey-gu** [comment]
We could later have some example/docs to highlight this sweet change :D

**Comment by haoxins** [comment]
> We could later have some example/docs to highlight this sweet change :D

I have another PR for ApplyEdge on the way.
BTW, I don't see the docs for Nebula client or SDKs :)

File: schema_manager.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+					return nil, fmt.Errorf("field type is different. "+
+						"Expected: %s, Actual: %s", expected.Type, actual.Type)
+				}
+				break
+			}
+		}
+		if !found {
+			// 4.2 Add the not exists field
+			q := expected.BuildAddTagFieldQL(tag.Name)
+			_, err := mgr.pool.ExecuteAndCheck(q)
```
Comment (Nicole00) [inline schema_manager.go:69]: If there are tens of new fields in expected schema, then we need to send tens of requests. can we append the statement and send just one request to server for add new fields?

**Comment by Nicole00** [comment]
> > We could later have some example/docs to highlight this sweet change :D
> 
> I have another PR for ApplyEdge on the way. BTW, I don't see the docs for Nebula client or SDKs :)

Great! Looking forward to your pr~  we have sdk api doc: https://pkg.go.dev/github.com/vesoft-inc/nebula-go/v3

File: schema_manager.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	if err != nil {
+		// 2. If the tag does not exist, create it
+		if strings.Contains(err.Error(), ErrorTagNotFound) {
+			return mgr.pool.CreateTag(tag)
+		}
+		return nil, err
+	}
+
+	// 3. If the tag exists, check if the fields are the same
+	fields, err := mgr.pool.DescTag(tag.Name)
```
[the remark this answers is not in the document]
Comment (haoxins) [reply schema_manager.go:47]: 👍 
done

File: schema_manager.go (hunk 1 above)
Comment (haoxins) [reply schema_manager.go:69]: good point

Comment (haoxins) [reply schema_manager.go]: done

File: schema_manager.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+		redundant := true
+		for _, expected := range tag.Fields {
+			if expected.Field == actual.Field {
+				redundant = false
+				break
+			}
+		}
+		if redundant {
+			// 5.1 Remove the not expected field
+			q := actual.BuildDropTagFieldQL(tag.Name)
```
Comment (Nicole00) [inline schema_manager.go:93]: ditto

**Review by Nicole00 — APPROVED** [review]
LGTM, Great sugar!

**Pull request merged**